flow/connectors/clickhouse/cdc.go

```diff
@@ -197,19 +197,34 @@ func (c *ClickHouseConnector) RenameTables(
 			c.logger.Info(fmt.Sprintf("table '%s' does not exist, skipping soft-deletes transfer for it", renameRequest.NewName))
 		}
 
-		// drop the dst table if exists
-		err = c.execWithLogging(ctx, "DROP TABLE IF EXISTS "+renameRequest.NewName)
-		if err != nil {
-			return nil, fmt.Errorf("unable to drop table %s: %w", renameRequest.NewName, err)
-		}
+		if c.dbSupportsExchange && originalTableExists {
+			c.logger.Info("attempting atomic exchange %s to %s", renameRequest.CurrentName, renameRequest.NewName)
+			// the clickhouse ATOMIC engine supports a special query to exchange two tables
+			// this means, we can have dependent (materialized) views and dictionaries on these tables.
+			err = c.execWithLogging(ctx, fmt.Sprintf("EXCHANGE TABLES %s and %s", renameRequest.NewName, renameRequest.CurrentName))
```

Comment:
there is a possibility of the resync table and the existing table having different schemas

would the EXCHANGE fail? would it fail if MVs depend on the schema of the existing table?

Reply:
So in my tests - the EXCHANGE itself works.

Case A: schema of _resync is compatible with the views (i.e. resync changes/drops/add columns not in View/Dictionary) everything works as expected.
Case B: schema of _resync impacts a dictionary: still works, however the dictionary won't update afterwards
Case C: schema of _resync impacts a view: EXCHANGE itself works, for materialized view the inserts of the data afterwards won't work because the MV references e.g. a nonexisting/incompatible column. Data insertion fails with Code 47.
![image](https://github.com/user-attachments/assets/4255b7d8-ebe7-42dc-a45b-727fbfb4c1cf)